### ISSUE TITLE
[BUG] Query cannot run in active session as long as the concurrent active session reach to 10

### ISSUE THREAD

**Issue by LantaoJin** [opensearch-project/opensearch-spark#876]
**What is the bug?**
Got follow error message when submit query in an active session:
{ "status": 429, "error": { "type": "ConcurrencyLimitExceededException", "reason": "Too Many Requests", "details": "domain concurrent active session can not exceed 10" } }

This is a bug to fix, since we should check the limit in create session instead of submitting query in active session.

**Comment by noCharger** [comment]
This is expected behavior. Please ref
https://github.com/opensearch-project/sql/pull/2390 for details

**Comment by LantaoJin** [comment]
@noCharger no. I don't think so. The limitation implementation in my option is not correct. We should limit the concurrent active sessions, for example to 10. But when there are 10 concurrent active sessions, all the 10 active sessions won't work as expected. For example, you are open an active session and submit queries in the only one session you have. Now someone else create 9 sessions. You will find your active active session (actually all 10 active sessions) become "inactive" suddenly.

**Comment by ykmr1224** [comment]
@LantaoJin +1. So the `leaseManager.borrow` should be called when creating new session.
